src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -0,0 +1,112 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.jenkins.pluginhealth.scoring.config.GithubConfiguration;
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.sun.istack.NotNull;
+import org.kohsuke.github.GHOrganization;
+import org.kohsuke.github.GHRateLimit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * Analyzes the {@link Plugin#getScm()} value to determine if the link is valid or not.
+ */
+@Component
+@Order(value = SCMLinkValidationProbe.ORDER)
+public final class SCMLinkValidationProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SCMLinkValidationProbe.class);
+    private static final String GH_REGEXP = "https://(?<server>[^/]*)/(?<repo>jenkinsci/[^/]*)(?:/.*)??";
+    private static final Pattern GH_PATTERN = Pattern.compile(GH_REGEXP);
+
+    public static final int ORDER = 1;
+
+    private final GithubConfiguration githubConfiguration;
+
+    public SCMLinkValidationProbe(GithubConfiguration githubConfiguration) {
+        this.githubConfiguration = githubConfiguration;
+    }
+
+    @Override
+    public ProbeResult doApply(Plugin plugin) {
+        if (plugin.getScm() == null || plugin.getScm().isBlank()) {
+            LOGGER.warn("{} has no SCM link", plugin.getName());
+            return ProbeResult.failure(key(), "No SCM link");
+        }
+        return fromSCMLink(plugin.getScm())
+            .map(repo -> ProbeResult.success(key(), "SCM link is valid"))
+            .orElseGet(() -> ProbeResult.failure(key(), "SCM link is invalid"));
+    }
+
+    @Override
+    public String key() {
+        return "scm";
+    }
+
+    @Override
+    protected boolean requiresRelease() {
+        return true;
+    }
+
+    private Optional<GHRepository> fromSCMLink(@NotNull String scm) {
+        Matcher matcher = GH_PATTERN.matcher(scm);
+        if (!matcher.find()) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+            }
+            return Optional.empty();
+        }
+        try {
+            final GitHub gitHub = this.getGitHub();
+            if (LOGGER.isTraceEnabled()) {
+                GHRateLimit rateLimit = gitHub.getRateLimit();
+                LOGGER.trace("GitHub rate: {}/{}, reset: {}", rateLimit.getRemaining(), rateLimit.getLimit(), rateLimit.getResetDate());
+            }
+            final GHRepository repo = gitHub.getRepository(matcher.group("repo"));
```

Comment:
What do we mean by "repo" here? What's the ideal value for it?
I can see that if it's null then the scm is invalid.

Reply:
The `repo` group is coming from the `GH_REGEXP`